website/content/v1.2/talos-guides/install/virtualized-platforms/vagrant-libvirt.md

```diff
@@ -0,0 +1,283 @@
+Current machine states:
+
+control-plane-node-1      running (libvirt)
+control-plane-node-2      running (libvirt)
+control-plane-node-3      running (libvirt)
+worker-node-1             running (libvirt)
+```
+
+Find out the IP addresses assigned by the libvirt DHCP by running:
+
+```bash
+virsh list | grep vagrant | awk '{print $2}' | xargs -t -L1 virsh domifaddr
+```
+
+Output will look like the following:
+
+```text
+virsh domifaddr vagrant_control-plane-node-2
+ Name       MAC address          Protocol     Address
+-------------------------------------------------------------------------------
+ vnet0      52:54:00:f9:10:e5    ipv4         192.168.121.119/24
+
+virsh domifaddr vagrant_control-plane-node-1
+ Name       MAC address          Protocol     Address
+-------------------------------------------------------------------------------
+ vnet1      52:54:00:0f:ae:59    ipv4         192.168.121.203/24
+
+virsh domifaddr vagrant_worker-node-1
+ Name       MAC address          Protocol     Address
+-------------------------------------------------------------------------------
+ vnet2      52:54:00:6f:28:95    ipv4         192.168.121.69/24
+
+virsh domifaddr vagrant_control-plane-node-3
+ Name       MAC address          Protocol     Address
+-------------------------------------------------------------------------------
+ vnet3      52:54:00:03:45:10    ipv4         192.168.121.125/24
+```
+
+Our control plane nodes have the IPs: `192.168.121.203`, `192.168.121.119`, `192.168.121.125` and the worker node has the IP `192.168.121.69`.
+
+Now you should be able to interact with Talos nodes that are in maintenance mode:
+
+```bash
+talosctl -n 192.168.121.203 disks --insecure
+```
+
+Sample output:
+
+```text
+DEV        MODEL   SERIAL   TYPE   UUID   WWID   MODALIAS                    NAME   SIZE     BUS_PATH
+/dev/vda   -       -        HDD    -      -      virtio:d00000002v00001AF4   -      8.6 GB   /pci0000:00/0000:00:03.0/virtio0/
+```
+
+## Installing Talos
+
+Pick an endpoint IP in the `vagrant-libvirt` subnet but not used by any nodes, for example `192.168.121.100`.
+
+Generate a machine configuration:
+
+```bash
+talosctl gen config my-cluster https://192.168.121.100:6443 --install-disk /dev/vda
+```
+
+Edit `controlplane.yaml` to add the virtual IP you picked to a network interface under `.spec.machine.network.interfaces`, for example:
+
+```yaml
+spec:
+  machine:
+    network:
+      interfaces:
+        - interface: eth0
+          dhcp: true
+          vip:
+            ip: 192.168.121.100
+```
+
+Apply the configuration to the initial control plane node:
+
+```bash
+talosctl -n 192.168.121.203 apply-config --insecure --file controlplane.yaml
+```
+
+You can tail the logs of the node:
+
+```bash
+sudo tail -f /tmp/control-plane-node-1.log
+```
+
+Set up your shell to use the generated talosconfig and configure its endpoint:
+
+```bash
+export TALOSCONFIG=$(realpath ./talosconfig)
+talosctl config endpoint 192.168.121.100
```

Comment:
this might be a problem Artem was referring to: it's better to use real IPs here, `talosctl config endpoint 192.168.121.119 192.168.121.125 192.168.121.69`

Reply:
Ah right, that makes total sense. I updated it.